pkg/storage/unified/apistore/util.go

```diff
@@ -31,6 +31,25 @@ func toListRequest(k *resource.ResourceKey, opts storage.ListOptions) (*resource
 		NextPageToken: predicate.Continue,
 	}
 
+	if opts.ResourceVersion != "" {
+		rv, err := strconv.ParseInt(opts.ResourceVersion, 10, 64)
+		if err != nil {
+			return nil, predicate, apierrors.NewBadRequest(fmt.Sprintf("invalid resource version: %s", opts.ResourceVersion))
+		}
+		req.ResourceVersion = rv
+	}
+
+	switch opts.ResourceVersionMatch {
+	case "", metav1.ResourceVersionMatchNotOlderThan:
+		req.VersionMatch = resource.ResourceVersionMatch_NotOlderThan
+	case metav1.ResourceVersionMatchExact:
+		req.VersionMatch = resource.ResourceVersionMatch_Exact
+	default:
+		return nil, predicate, apierrors.NewBadRequest(
+			fmt.Sprintf("unsupported version match: %v", opts.ResourceVersionMatch),
+		)
+	}
+
```

Comment:
Question: is there any special reason why you moved this up? Did you see some side-effects or is it just logical re-ordering?

Reply:
Yes this was necessary because the early return of the [history/trash logic](https://github.com/grafana/grafana/pull/102417/files#diff-535b47a5e828f38f3b4db21085f8b9463e587b9981775d2a909f47e948a826a5R91) results in this code not being executed. So we would not propagate the information to the downstream request. Hence it's necessary to move this up ahead.

Reply:
I would've done the same with `if opts.Predicate.Field != nil && !opts.Predicate.Field.Empty() {`, but I do not understand the code enough and we're also partially clearing some of the properties like `req.Options.Fields`, so I wasn't sure.